torchtune/utils/ppo_utils/_generation.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+import torch
+from torchtune.utils._generation import sample, update_stop_tokens_tracker
+
+
+def generate_next_token(
```

Comment:
Is the only difference between this and the existing `generate_next_token` the inclusion of the `mask` parameter? If so I'd be inclined to just add it to the existing function